packages/dom/src/lib/ElementAssertion.ts

```diff
@@ -158,20 +158,12 @@ export class ElementAssertion<T extends Element> extends Assertion<T> {
    */
 
   public toHaveStyle(expected: Partial<CSSStyleDeclaration>): this {
-    if (!this.actual.ownerDocument.defaultView) {
-      throw new Error("The element is not attached to a document with a default view.");
-    }
-    if (!(this.actual instanceof HTMLElement)) {
-      throw new Error("The element is not an HTMLElement.");
-    }
-
-    const window = this.actual.ownerDocument.defaultView;
 
-    const received = window.getComputedStyle(this.actual);
+    const [expectedStyle, receivedStyle] = getExpectedAndReceivedStyles(this.actual, expected);
 
-    const { props, expectedStyle } = normalizeStyles(expected);
-
-    const receivedStyle = getReceivedStyle(props, received);
+    if (!expectedStyle || !receivedStyle) {
+      throw new Error("No available styles.");
```

Comment:
Maybe we can improve the messaging in here:

```suggestion
      throw new Error("Currently there are no available styles.");
```